tools/richlist.js

```diff
@@ -0,0 +1,559 @@
+            return results[index] || {};
+          }).forEach(function (result, i) {
+            var code = batch.requests[i].format ? batch.requests[i].format(result.result) : result.result;
+            var account = batch.requests[i].params[0];
+            data[account] = { address: account };
+            if (code.length > 2) {
+              // 0: normal address, 1: contract
+              data[account].type = 1; // contract case
+            }
+
+          });
+          innerCallback(null, data);
+        });
+      }, function(data, innerCallback) {
+        // batch rpc job
+        var batch = web3.createBatch();
+        for (var i = 0; i < accounts.length; i++) {
+          var account = accounts[i];
+          batch.add(web3.eth.getBalance.request(account));
+        }
+
+        batch.requestManager.sendBatch(batch.requests, function(err, results) {
+          if (err) {
+            console.log("ERROR: fail to getBalance batch job:", err);
+            innerCallback(err);
+            return;
+          }
+          results = results || [];
+          batch.requests.map(function (request, index) {
+            return results[index] || {};
+          }).forEach(function (result, i) {
+            var balance = batch.requests[i].format ? batch.requests[i].format(result.result) : result.result;
+
+            let ether;
+            if (typeof balance === 'object') {
+              ether = parseFloat(balance.div(1e18).toString());
+            } else {
+              ether = balance / 1e18;
+            }
+            data[batch.requests[i].params[0]].balance = ether;
+          });
+          innerCallback(null, data);
+        });
+      }], function(err) {
+        callback(err, data, lastAccount);
+      });
+    }
+  ], function(error, accounts, lastAccount) {
+    if (error) {
+      console.log(error);
+      process.exit(9);
+      return;
+    }
+
+    //console.log(JSON.stringify(accounts, null, 2));
+    offset = lastAccount;
+    let j = Object.keys(accounts).length;
+    self.index += j;
+    console.log(' * ' + j + ' / ' + self.index + ' accounts, offset = ' + offset);
+    if (updateCallback) {
+      updateCallback(accounts, blockNumber);
+    }
+    setTimeout(function() {
+      makeParityRichList(number, lastAccount, blockNumber, updateCallback);
+    }, 300);
+  });
+}
+
+/**
+ * Write accounts to DB
+ */
+var updateAccounts = function(accounts, blockNumber) {
+  // prepare
+  var bulk = Object.keys(accounts).map(function(j) {
+    let account = accounts[j];
+    account.blockNumber = blockNumber;
+    return account;
+  });
+
+  bulkInsert(bulk);
+}
+
+var bulkInsert = function(bulk) {
+  if (!bulk.length) {
+    return;
+  }
+
+  var localbulk;
+  if (bulk.length > 300) {
+    localbulk = bulk.splice(0, 200);
+  } else {
+    localbulk = bulk.splice(0, 300);
+  }
+  Account.collection.insert(localbulk, function(error, data) {
+    if (error) {
+      if (error.code == 11000) {
```

Comment:
Maybe add a comment on why this `error.code` specifically

Reply:
That's cool, would you mind add a comment above line 392 saying so? Something small like `// For already exists case`